### PR TITLE
Add quality and timestamp on monitor suscriptions

### PR DESCRIPTION
**Pull request by bajcmartinez**
#39 - Get the quality and timestamp

### CODE REVIEW

**Comment by jmbeach** [comment]
Can you add XML comments to the new public fields you added? See the appveyor output https://ci.appveyor.com/project/Hyla-Soft-Inc/h-opc/build/1.0.141

**Comment by jmbeach** [comment]
There are still issues from the build https://ci.appveyor.com/project/Hyla-Soft-Inc/h-opc/build/1.0.142. Please follow the build instructions in the readme.

**Comment by bajcmartinez** [comment]
Yes, sorry, I missed some of the elements, and was getting some warnings, I think they all should be fixed now.

File: h-opc/Da/DaClient.cs [hunk 1]
```diff
@@ -184,7 +185,13 @@ public void Monitor<T>(string tag, Action<T, Action> callback)
       {
         T casted;
         TryCastResult(values[0].Value, out casted);
-        callback(casted, unsubscribe);
+        var monitorEvent = new MonitorEvent<T>();
+        monitorEvent.Value = casted;
+        monitorEvent.SourceTimestamp = values[0].Timestamp;
+        monitorEvent.ServerTimestamp = values[0].Timestamp;
```
Comment (jmbeach) [inline h-opc/Da/DaClient.cs]: duplicate line

File: tests/App.config [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8" ?>
 <configuration>
   <appSettings>
-    <add key="UATestEndpoint" value="opc.tcp://localhost:61210/UA/SampleClient"/>
+    <add key="UATestEndpoint" value="opc.tcp://md1m8frc:51210/UA/SampleServer"/>
```
Comment (jmbeach) [inline tests/App.config]: Don't want this change.
Comment to myself: This file probably should not be tracked to begin with. Should make a configurable template.

**Comment by jmbeach** [comment]
I'm very sorry that it's taken me so long to respond. Please take a look at the minor changes I commented on [this commit](https://github.com/hylasoft-usa/h-opc/pull/57/commits/8686c8356af95c3bafe7b35eebf62b5950233570). I like this addition a lot. Thanks for your contribution.

File: README.md [hunk 3]
```diff
@@ -109,9 +109,11 @@ client.Write("path.to.num", 42);
 Dead-simple monitoring:
 
 ````cs
-client.Monitor<string>("path.to.string", (newValue, unsubscribe) =>
+client.Monitor<string>("path.to.string", (readEvent, unsubscribe) =>
 {
-  DoSomethingWithYourValue(newValue);
+  DoSomethingWithQuality(readEvent.Quality);
```
Comment (jmbeach) [inline README.md]: please remove quality and timestamp from example. This is supposed to be the simplest use-case where the developer just wants to use a value.

**Comment by bajcmartinez** [comment]
Hi, sorry it took me so long to reply, I was on holidays. The requested change is now on the PR. Thanks

File: README.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-h-opc [![Build status](https://ci.appveyor.com/api/projects/status/oajkgccisoe98gip?svg=true)](https://ci.appveyor.com/project/Hyla-Soft-Inc/h-opc) [![NuGet Status](http://img.shields.io/nuget/v/H.Opc.svg)](https://www.nuget.org/packages/H.Opc) [![Coverage Status](https://coveralls.io/repos/github/jmbeach/h-opc/badge.svg?branch=master)](https://coveralls.io/github/jmbeach/h-opc?branch=master)
+bajcmartinez:add-quality-and-timestamp-on-monitor-suscriptionsh-opc [![Build status](https://ci.appveyor.com/api/projects/status/oajkgccisoe98gip?svg=true)](https://ci.appveyor.com/project/Hyla-Soft-Inc/h-opc) [![NuGet Status](http://img.shields.io/nuget/v/H.Opc.svg)](https://www.nuget.org/packages/H.Opc) [![Coverage Status](https://coveralls.io/repos/github/jmbeach/h-opc/badge.svg?branch=master)](https://coveralls.io/github/jmbeach/h-opc?branch=master)
```
Comment (jmbeach) [inline README.md]: Don't want this change

**Comment by jmbeach** [comment]
Thanks for the latest commit. There's still an issue with the README and the app.config. Please take a look. I look forward to integrating your changes into the repository.

**Comment by jmbeach** [comment]
@bajcmartinez I hope I didn't discourage you from committing. I feel like this change is valuable if you still want to submit it. Thank you.

**Comment by bajcmartinez** [comment]
Hi, sorry, I was super busy this last few weeks as I was on holidays and then on the process to move to another country, I'll check the issues and prepare a fix ASAP. Thanks

**Comment by jmbeach** [comment]
Thanks for your submission!

**Pull request merged**